tests/test_ops_syntax.py

```diff
@@ -1064,3 +1065,31 @@ def foo1_bar_impl(x):
 
         # Without the inner handler, foo1.bar should also call Foo.bar
         assert foo1.bar(42) == f"Foo.bar({foo1}, 42)"
+
+
+def test_operation_dataclass():
+    @dataclasses.dataclass
+    class Point:
+        x: int
+        y: int
+
+    @defdata.register(Point)
+    class PointTerm(Term[Point], Point, metaclass=DataclassTermMeta):
+        pass
```

Comment:
I wonder if we could expose a decorator you can slap onto the dataclass similar to `@jax.tree_util.register_dataclass` that just makes the type known to effectful. Then the user can omit this complicated `PointTerm` definition entirely. That also keeps our options open in case we ever wanted to implement this through some other avenue than a metaclass.